cmdLine/src/test/java/topgun/cmdline/TestFileParser.java

```diff
@@ -0,0 +1,121 @@
+package topgun.cmdline;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.kohsuke.args4j.CmdLineException;
+import org.kohsuke.args4j.CmdLineParser;
+
+import java.io.File;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Paths;
+
+public class TestFileParser {
+
+    String[] param = null;
+    JfrParseCommandLine options = null;
+    CmdLineParser parser = null;
+    Configuration configuration = null;
+
+    @Before
+    public void setup() throws URISyntaxException {
+
+        param = new String[]{
+                "--jfr",
+                getAbsolutePath(),
+                "--outDir",
+                "../",
+                "--userCodePackagePrefix",
+                "com.tradeengine.trade.engine"
+        };
+        options = new JfrParseCommandLine();
+        parser = new CmdLineParser(options);
+        try {
+            parser.parseArgument(param);
+
+        } catch (CmdLineException e) {
+            e.printStackTrace();
+            return;
+        }
+        options.validate(parser);
+        configuration = new Configuration((options));
+    }
+
+    //Getting the Absolute Path for the
+    private String getAbsolutePath() throws URISyntaxException {
+        URL res = getClass().getClassLoader().getResource("jfr_files/flight_recording_1108comintellijideaMain4564_2020-11-30_12-48-18_ae084800.jfr");
+        File file = Paths.get(res.toURI()).toFile();
+        return file.getAbsolutePath();
+    }
+
+    @After
+    public void cleanUp() {
+
+    }
+
+    //Test to check the consumed Cpu Events
+    @Test
+    public void test_parse_method() {
+        Totals totals = new Totals();
+        new FileParser(options.jfr(), options, totals, configuration).parse();
+        Assert.assertEquals(134, totals.consumedCpuEvents().get());
+    }
+
+    //Test to check the Ignored Thread Allocation Events
+    @Test
+    public void ignoreThread() {
```

Comment:
you can combine several tests together. You can test more than one thing in a test